src/App.jsx

```diff
@@ -14,33 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortField, setSortField] = useState('');
+  const [showReset, setShowReset] = useState(false);
+  const [reverse, setReverse] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const toSort = field => {
+    let sortedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (field) {
+      case 'Sort alphabetically':
+        sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case 'Sort by length':
+        sortedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      case 'Reverse':
+        sortedGoods.reverse();
+        setReverse(!reverse);
+        break;
+      case 'Reset':
+        sortedGoods = [...goodsFromServer];
+        break;
+      default:
+        break;
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setGoods(sortedGoods);
+  };
+
+  const handleSort = field => {
+    setSortField(field);
+    toSort(field);
+
+    if (field === 'Reset') {
+      setShowReset(false);
+      setReverse(false);
+    } else {
+      setShowReset(true);
+    }
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {['Sort alphabetically', 'Sort by length'].map(field => (
```

Comment:
if you'd like to use an array here you can create an array of objects

```
const sortFieldsArray = [
  {
    text: 'Sort alphabetically',
    value: 'alphabetically'
  },
  {
    text: 'Sort by length',
    value: 'length'
  },
]
```

```suggestion
        {sortFieldsArray.map(({ text, value })=> (
```